### PR TITLE
HM-2834: Hide registered sellers

### PR DESCRIPTION
**Pull request by dta-tyler**
This pull request hides the profile page of a registered (unapproved) seller from a buyer when accessed using a direct URL, by instead displaying a 404 page. It still allows a seller to view their own profile.

### CODE REVIEW

**Comment by dta-tyler** [comment]
Logic has been updated to display profiles for recruiter/hybrid sellers without an approved category. Users of type `admin` and `assessor` can always view a seller profile.

File: app/main/views/suppliers.py [hunk 1]
```diff
@@ -33,6 +33,14 @@ def get_supplier(code):
 
     supplier = DataAPIClient().get_supplier(code)['supplier']
 
+    try:
+        if supplier['recruiter'] == 'no' and len(supplier['domains']['assessed']) == 0:
+            if ((current_user.role != 'supplier' and current_user.supplier_code != code) and
+               current_user.role not in ['admin', 'assessor']):
```
Comment (tobiasbrown) [inline app/main/views/suppliers.py:39]: I think this can be simplified if you merge all the conditions into the first `if` statement.

Comment (dta-tyler) [reply app/main/views/suppliers.py]: Yes, it can be simplified. I have pushed a change to rectify this.

**Review by dta-tom — APPROVED** [review]
Looks good

**Comment by dta-tyler** [comment]
This feature will not be released as Digital Marketplace is moving to BuyICT.

**Pull request closed without merging**